package/MDAnalysis/analysis/polymer.py

```diff
@@ -0,0 +1,170 @@
+
+
+class PersistenceLength(AnalysisBase):
+    r"""Calculate the persistence length for polymer chains
+
+    The persistence length is the length at which two points on the polymer
+    chain become decorrelated.
+
+    Notes
+    -----
+    This analysis requires that the trajectory supports indexing
+
+    .. versionadded:: 0.12.0
+    """
+    def __init__(self, atomgroups,
+                 start=None, stop=None, step=None):
+        """Calculate the persistence length for polymer chains
+
+        Parameters
+        ----------
+        atomgroups : list
+            List of atomgroups.  Each atomgroup should represent a single
+            polymer chain, ordered in the correct order.
+        start : int, optional
+            First frame of trajectory to analyse, Default: 0
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        step : int, optional
+            Step between frames to analyse, Default: 1
+        """
+        self._atomgroups = atomgroups
+
+        # Check that all chains are the same length
+        lens = [len(ag) for ag in atomgroups]
+        chainlength = len(atomgroups[0])
+        if not all([l == chainlength for l in lens]):
+            raise ValueError("Not all AtomGroups were the same size")
+
+        self._setup_frames(atomgroups[0].universe.trajectory,
+                           start, stop, step)
+
+        self._results = np.zeros(chainlength - 1, dtype=np.float32)
+
+    def _single_frame(self):
+        # could optimise this by writing a "self dot array"
+        # we're only using the upper triangle of np.inner
+        # function would accept a bunch of coordinates and spit out the
+        # decorrel for that 
+        n = len(self._atomgroups[0])
+
+        for chain in self._atomgroups:
+            # Vector from each atom to next
+            vecs = chain.positions[1:] - chain.positions[:-1]
+            # Normalised to unit vectors
+            vecs /= np.sqrt((vecs * vecs).sum(axis=1))[:, None]
+
+            inner_pr = np.inner(vecs, vecs)
+            for i in xrange(n-1):
+                self._results[:(n-1)-i] += inner_pr[i, i:]
+
+    def _normalise(self):
+        n = len(self._atomgroups[0])
+
+        norm = np.linspace(n - 1, 1, n - 1)
+        norm *= len(self._atomgroups) * self.nframes
+
+        self.results = self._results / norm
+        self._calc_bond_length()
+
+    def _calc_bond_length(self):
+        """calculate average bond length"""
+        bs = []
+        for ag in self._atomgroups:
+            pos = ag.positions
+            b = calc_bonds(pos[:-1], pos[1:]).mean()
+            bs.append(b)
+        self.lb = np.mean(bs)
+
+    def perform_fit(self):
+        """Fit the results to an exponential decay"""
+        from scipy.optimize import curve_fit
+
+        try:
+            results = self.results
+        except AttributeError:
+            raise ValueError("Use the run method first")
```

Comment:
In other cases such as [rms.RMSD.save()](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/analysis/rms.py#L489) we used `NoDataError` because it expressed most concisely what the problem is. The error message then says how to fix it.

(Admittedly, other classes might still use other exceptions but I think we should standardize on `NoDataError` and we might as well continue here...)